server/app/services/question/ActiveAndDraftQuestions.java

```diff
@@ -68,6 +68,7 @@ private ActiveAndDraftQuestions(Version active, Version draft, Version withDraft
 
     versionedByName =
         Sets.union(activeNameToQuestion.keySet(), draftNameToQuestion.keySet()).stream()
+            .sorted()
```

Comment:
Was this call necessary? Does this mean there's now an implication of sort order that wasn't there before?

Mostly wondering since we're taking a stance in  https://github.com/civiform/civiform/issues/3029 that sorting would be done client side.

Additionally, the type we're assigning this to (e.g. `Map`) has no explicit guarantees about order. Given this, it feels like the call may be unnecessary.

Reply:
ImmutableMap guaranteers stable order (insertion order). I added this as I saw some flakiness in the test where there were 2 questions. Sometimes their order changed. Given that you are adding sort-by-recent I decided that sorting here by name is ok and short-lived anyway.

Reply:
I guess my general comment is that a map is an unordered type (even if the concrete implementation preserves order). Given that this is short-lived, it sounds good to me. What do you think about addding a TODO against #3029 and we can remove it at that point?

Or maybe I'm just being overly pedantic about assuming order.

Reply:
Done.